trikot-streams/streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/CombineLatestProcessorExtensions.kt

```diff
@@ -115,8 +115,8 @@ fun <T, R1, R2> Publisher<T>.safeCombine(publisher1: Publisher<R1>, publisher2:
     return (this as Publisher<Any>).combine(listOf(publisher1, publisher2) as List<Publisher<Any>>)
         .filter { list ->
             list[0] as? T != null &&
-                list[1] as? T != null &&
-                list[2] as? T != null
+                list[1] as? R1 != null &&
```

Comment:
Nice catch, and probably a result of some wild type erasure magic happening here for that to work with the wrong cast.